src/wp-includes/blocks/image.php

```diff
@@ -30,14 +31,278 @@ function render_block_core_image( $attributes, $content ) {
 		$processor->set_attribute( 'data-id', $attributes['data-id'] );
 	}
 
+	$lightbox_enabled  = false;
+	$link_destination  = isset( $attributes['linkDestination'] ) ? $attributes['linkDestination'] : 'none';
+	$lightbox_settings = block_core_image_get_lightbox_settings( $block->parsed_block );
+
+	// If the lightbox is enabled and the image is not linked, flag the lightbox to be rendered.
+	if ( isset( $lightbox_settings ) && 'none' === $link_destination ) {
+
+		if ( isset( $lightbox_settings['enabled'] ) && true === $lightbox_settings['enabled'] ) {
+			$lightbox_enabled = true;
+		}
+	}
+
+	// If at least one block in the page has the lightbox, mark the block type as interactive.
+	if ( $lightbox_enabled ) {
+		$block->block_type->supports['interactivity'] = true;
+	}
+
+	// Determine whether the view script should be enqueued or not.
+	$view_js_file = 'wp-block-image-view';
+	if ( ! wp_script_is( $view_js_file ) ) {
+		$script_handles = $block->block_type->view_script_handles;
+
+		// If the script is not needed, and it is still in the `view_script_handles`, remove it.
+		if ( ! $lightbox_enabled && in_array( $view_js_file, $script_handles, true ) ) {
+			$block->block_type->view_script_handles = array_diff( $script_handles, array( $view_js_file ) );
+		}
+		// If the script is needed, but it was previously removed, add it again.
+		if ( $lightbox_enabled && ! in_array( $view_js_file, $script_handles, true ) ) {
+			$block->block_type->view_script_handles = array_merge( $script_handles, array( $view_js_file ) );
+		}
+	}
+
+	if ( $lightbox_enabled ) {
+		return block_core_image_render_lightbox( $processor->get_updated_html(), $block->parsed_block );
+	}
+
 	return $processor->get_updated_html();
 }
 
+/**
+ * Add the lightboxEnabled flag to the block data.
+ *
+ * This is used to determine whether the lightbox should be rendered or not.
+ *
+ * @param  array $block Block data.
+ * @return array        Filtered block data.
+ */
+function block_core_image_get_lightbox_settings( $block ) {
+	// Get the lightbox setting from the block attributes.
+	if ( isset( $block['attrs']['lightbox'] ) ) {
+		$lightbox_settings = $block['attrs']['lightbox'];
+		// If the lightbox setting is not set in the block attributes,
+		// check the legacy lightbox settings that are set using the
+		// `gutenberg_should_render_lightbox` filter.
+		// We can remove this elseif statement when the legacy lightbox settings are removed.
+	} elseif ( isset( $block['legacyLightboxSettings'] ) ) {
+		$lightbox_settings = $block['legacyLightboxSettings'];
+	}
+
+	if ( ! isset( $lightbox_settings ) ) {
+		$lightbox_settings = gutenberg_get_global_settings( array( 'lightbox' ), array( 'block_name' => 'core/image' ) );
```

Comment:
PR to remove these `gutenberg` prefixes [here](https://github.com/WordPress/gutenberg/pull/54678)

Reply:
> This has to be updated in Gutenberg because the files are auto-generated.

Oh, that's right. I forgot about that, sorry 🤦‍♂️ 

I'll make the PR and mark it with the backport label in Gutenberg.

Reply:
I made a PR and merged it already! See link above.

Reply:
Oh wait you mean for the view files? I didn't do a PR for those

Reply:
I mean the one with the `@global` comments: https://github.com/luisherranz/wordpress-develop/commit/3eb79354bcaeef9e84b3cdc3f69a5ef1a1250bcd

Reply:
The view files are fine, as long as this we include this change here: https://github.com/luisherranz/wordpress-develop/commit/facb976a29a5d968031558c27845daea5b480a1a

Reply:
I added the missing `@globals` in Gutenberg:

- https://github.com/WordPress/gutenberg/pull/54683

Reply:
Got it, thanks!